### PR TITLE
Xsiam jira integration

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XDR-52062

## Description
New integration to fetch audit records from Jira.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Jira/Integrations/JiraAuditRecords/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.
+After you successfully execute a command, a DBot message appears in the War Room with the command details.
+
+### fetch-events
+
+***
+
+#### Base Command
+
+`fetch-events`
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/README.md:35]:
```suggestion
`jira-fetch-events`
```

File: Packs/Jira/Integrations/JiraAuditRecords/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+This integration was integrated and tested with version 3 of Jira Audit Records rest API
+
+## Configure Jira Audit Records on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for Jira Audit Records.
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/README.md:6]:
```suggestion
2. Search for Jira Events Collector.
```

File: Packs/Jira/ReleaseNotes/2_0_0.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### New: Jira Audit Records
+- The integration fetches Audit Records form Jira (Available XSIAM).
```
Comment (yaakovi) [inline Packs/Jira/ReleaseNotes/2_0_0.md:4]:
```suggestion
- The integration fetches Audit Records from Jira (Available in Cortex XSIAM).
```

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
+category: Analytics & SIEM
+commonfields:
+  id: JiraAuditRecords
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:3]: Change integration id, name, and display to `Jira Events Collector`.
Also, Update the folder/file names.

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  version: -1
+configuration:
+- display: Incident type
+  name: incidentType
+  required: false
+  type: 13
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:9]:
```suggestion
```

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  name: method
+  required: false
+  options:
+  - GET
+  - POST
+  - PUT
+  - HEAD
+  - PATCH
+  - DELETE
+  type: 15
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:26]: Why do you need it? Seems redundant.

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  name: proxy
+  required: false
+  type: 8
+- defaultvalue: 3 days
+  display: First fetch time (<number> <time unit>, e.g., 12 hours, 1 day, 3 months). default is 3 days.
+  name: first_fetch
+  required: true
+  type: 0
+- defaultvalue: 100
+  display: The maximum number of incidents per fetch. Default is 100.
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:46]:
```suggestion
  display: The maximum number of events per fetch. Default is 100.
```

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  name: max_fetch
+  required: true
+  type: 0
+description: '[Enter a comprehensive, yet concise, description of what the integration
+  does, what use cases it is designed for, etc.]'
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:51]: update

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+display: Jira Audit Records
+name: JiraAuditRecords
+script:
+  commands:
+  - deprecated: false
+    description: ''
+    execution: false
+    name: fetch-events
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:59]:
```suggestion
    name: jira-fetch-events
```

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+      isArray: true
+      name: max_fetch
+      required: false
+      secret: false
+    - description: First fetch time (<number> <time unit>, e.g., 12 hours, 1 day, 3 months). default is 3 days.
+      isArray: true
+      name: first_fetch
+      required: false
+      secret: false
+    outputs:
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.yml:71]: command should not have outputs.

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+
+    if command == 'test-module':
+        get_events.run(max_fetch=1)
+        demisto.results('ok')
+
+    elif command in ('fetch-events', 'JiraAuditRecords-get-events'):
+        events = get_events.run(int(demisto_params.get('max_fetch', 100)))
+
+        if events:
+            demisto.setLastRun(get_events.set_next_run(events[0]))
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py:203]: Are the events sorted?

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+            if command == 'fetch-events':
+                send_events_to_xsiam(events, 'Jira', 'Audit Records')
+            else:
+                get_events.events_to_incidents(events)
+                command_results = CommandResults(
+                    readable_output=tableToMarkdown('Jira Audit Records', events, removeNull=True, headerTransform=pascalToSpace),
+                    outputs_prefix='JiraAudit.Records',
+                    outputs_key_field='id',
+                    outputs=events,
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py:212]: command shouldn't return outputs
```suggestion
```

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+
+        for event in events:
+            incident = {
+                'name': f"Jira Audit Records - {event['summary']} - {event['id']}",
+                'occurred': event.get('created', '').removesuffix('+0000') + 'Z',
+                'rawJSON': json.dumps(event)
+            }
+            incidents.append(incident)
+
+        demisto.incidents(incidents)
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py:148]: Remove

**Review by yaakovi — CHANGES_REQUESTED** [review]
update integration logo

File: Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py (hunk 11 above)
Comment (MosheEichler) [reply Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py:203]: Yes
https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/#:~:text=Responses%20are%20listed%20in%20ascending%20order%20by%20default

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py [hunk 14]
```diff
@@ -196,20 +182,17 @@ def main():
         get_events.run(max_fetch=1)
         demisto.results('ok')
 
-    elif command in ('fetch-events', 'JiraAuditRecords-get-events'):
+    elif command in ('fetch-events', 'jira-fetch-events'):
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py:185]: Sorry about that, My bad.
```suggestion
    elif command in ('fetch-events', 'jira-get-events'):
```

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py [hunk 15]
```diff
@@ -196,20 +182,17 @@ def main():
...
         events = get_events.run(int(demisto_params.get('max_fetch', 100)))
 
         if events:
             demisto.setLastRun(get_events.set_next_run(events[0]))
             if command == 'fetch-events':
                 send_events_to_xsiam(events, 'Jira', 'Audit Records')
+                demisto.updateModuleHealth({'eventsPulled': len(events)})
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py:192]: Already included in the `send_events_to_xsiam` function
```suggestion
```

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py [hunk 16]
```diff
@@ -196,20 +182,17 @@ def main():
         get_events.run(max_fetch=1)
         demisto.results('ok')
 
-    elif command in ('fetch-events', 'JiraAuditRecords-get-events'):
+    elif command in ('fetch-events', 'jira-fetch-events'):
         events = get_events.run(int(demisto_params.get('max_fetch', 100)))
 
         if events:
             demisto.setLastRun(get_events.set_next_run(events[0]))
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.py:189]: write to log the new last run (debug level)

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  required: true
+  type: 9
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml:23]: move to the end of the parameter list

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+description: '[Enter a comprehensive, yet concise, description of what the integration
+  does, what use cases it is designed for, etc.]'
+display: Jira Events Collector
+name: Jira Events Collector
+script:
+  commands:
+  - deprecated: false
+    description: 'Returns a list of a list of audit records'
+    execution: false
+    name: jira-fetch-events
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml:43]:
```suggestion
    name: jira-get-events
```

File: Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml [hunk 19]
```diff
@@ -0,0 +1,63 @@
+category: Analytics & SIEM
+commonfields:
+  id: Jira Events Collector
```
Comment (yaakovi) [inline Packs/Jira/Integrations/JiraEventsCollector/JiraEventsCollector.yml:3]: again, sorry about that.
here and everywhere.
```suggestion
  id: Jira Event Collector
```

**Review by yaakovi — APPROVED** [review]

File: Packs/Jira/ModelingRules/JiraEventCollector/JiraEventCollector.yml [hunk 20]
```diff
@@ -0,0 +1,5 @@
+name: JiraEventCollector
+id: JiraEventCollector
+fromversion: 3.3.0
+tags: a
```
Comment (ShahafBenYakir) [inline Packs/Jira/ModelingRules/JiraEventCollector/JiraEventCollector.yml:4]: remove

File: Packs/Jira/Integrations/JiraEventCollector/JiraEventCollector.yml [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+      isArray: true
+      name: first_fetch
+      required: false
+      secret: false
+  runonce: false
+  script: '-'
+  type: python
+  subtype: python3
+  dockerimage: demisto/fastapi:1.0.0.29406
+fromversion: 5.5.0
```
Comment (ShahafBenYakir) [inline Packs/Jira/Integrations/JiraEventCollector/JiraEventCollector.yml:59]:
```suggestion
fromversion: 6.5
```

**Review by richardbluestone — APPROVED** [review]

**Comment by richardbluestone** [comment]
I'm not sure whether the parameters should be:
Incident type > alert type
Your server URL > Server URL
Fetch Incidents > Fetch Alerts

**Pull request merged**